content/src/storage/FileSystemContentStorage.ts

```diff
@@ -53,19 +54,31 @@ export class FileSystemContentStorage implements ContentStorage {
     }
   }
 
-  private async retrieveWithEncoding(id: string, encoding: ContentEncoding | null): Promise<ContentItem | undefined> {
+  private async retrieveWithEncoding(
+    id: string,
+    encoding: ContentEncoding | null,
+    range?: ContentRange
+  ): Promise<ContentItem | undefined> {
     const extension = encoding ? '.' + encoding : ''
     const filePath = this.getFilePath(id) + extension
 
     if (await existPath(filePath)) {
       const stat = await fs.promises.stat(filePath)
+      if (range && stat.size) {
+        const safeRange = {
+          start: Math.min(range.start, stat.size - 1),
```

Comment:
shouldn't this be `Math.max(range.start, 0)` ?

Reply:
I made it that way initially to prevent having a `range.start` bigger than the file size. I've now refactored the code to make it clearer, setting some default values when the ranges are not provided and validating them later